src/platform/activity/activity.service.ts

```diff
@@ -185,9 +184,24 @@ export class ActivityService {
     const collaborationMap = new Map(collaborations.map(c => [c.id, c]));
 
     // Filter activities that have a corresponding collaboration
-    const activityData = activities
-      .filter(activity => collaborationMap.has(activity.collaborationID))
-      .slice(0, limit);
+    const filteredActivities = activities.filter(activity =>
+      collaborationMap.has(activity.collaborationID)
+    );
+
+    // Create a map of collaboration IDs to latest activities
+    const latestActivityMap = new Map();
```

Comment:
I would only rename this to be `latestActivityPerJourneyMap` or `latestActivityPerCollaborationMap` because it will be easier to understand the code when we come back to it after some time.